### PR TITLE
DM-27225: Update dependencies and fix Docker upload

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

**Review by afausti — APPROVED** [review]
I was able to `make update` successfully on this branch and verify that `daf-butler` and `lsst-sphgeom` are built correctly.  When I run `explog --help`, for example, I can see that `daf_butler` is imported without errors. I can start the development webserver with `tox -e run` after setting  the envvar `BUTLER_URI_1` but I get an error if I do `explog run`.  That error does not seem to be related to the changes in this branch.  However, it is important to get the `explog run` command working because it is the way the application is started in the Dockerfile.

File: src/explog/app.py [hunk 1]
```diff
@@ -27,17 +27,31 @@ def create_app(**configs: typing.Any) -> web.Application:
...
+        explog_db = LogMessageDatabase(config.exposure_log_database_url)
+        root_app["explog/exposure_log_database"] = explog_db
+
+    async def cleanup(app: typing.Any):
+        explog_db = root_app["explog/exposure_log_database"]
+        await explog_db.close()
+
     root_app = web.Application()
+    root_app.on_startup.append(startup)
+    root_app.on_cleanup.append(cleanup)
```
Comment (afausti) [inline src/explog/app.py:53]: Looks good, glad that `on_startup()` and `on_cleanup()` did the trick.

**Review by afausti — APPROVED** [review]
LGTM

File: src/explog/app.py (hunk 1 above)
Comment (r-owen) [reply src/explog/app.py:53]: Thank you very much for your help!

**Pull request merged**